Documentation/designs/prometheus-operator-cli.md

```diff
@@ -0,0 +1,61 @@
+# Prometheus Operator CLI
+
+## Summary
+
+This document proposes a new CLI tool to manage Prometheus Operator resources, allowing users to create, update, and delete Prometheus Operator resources using a simple CLI interface as well as
+troubleshoot and debug Prometheus Operator resources.
+
+## Background
+
+During KubeCon Paris 2024, we received feedback from users that managing Prometheus Operator resources is difficult and error-prone. Users reported that they often encounter issues when creating, updating, and deleting Prometheus Operator resources, and that troubleshooting and debugging Prometheus Operator resources is challenging, below you can find some of the issues that users reported:
+
+### Why my Prometheus resource is not being created?
+
+People were struggling to troubleshoot why their Prometheus resources were not being created. They were not sure if the issue was with the Prometheus Operator or with the Prometheus resource itself.
+After long investigation, they found out that the Prometheus Operator were only watching the namespace where the Prometheus Operator was deployed, and the Prometheus resource was being created in a different namespace.
+
+### Why my ServiceMonitor is not being created?
+
+People were struggling to troubleshoot why their ServiceMonitor resources were not being created. They were not sure if the issue was with the Prometheus Operator or with the ServiceMonitor resource itself.
+
+After long investigation, they found out that the matchingLabels field in the ServiceMonitor resource was not matching the labels of the target Service.
+
+These are just some common issues that users reported, but there are many other issues that users face when managing Prometheus Operator resources, and we believe that a CLI tool can help users to manage Prometheus Operator resources more easily and efficiently.
+
+## Proposal
+
+We propose to create a new CLI tool to manage Prometheus Operator resources, allowing users to create, update, and delete Prometheus Operator resources using a simple CLI interface as well as troubleshoot and debug Prometheus Operator resources.
+
+The main idea is to distribute this CLI as a [kubectl plugin](https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/) and the use experience should be similar to the `kubectl` CLI, as you can see in the example below:
+
+```bash
+kubectl prometheus-operator create prometheus --name my-prometheus --namespace my-namespace --replicas 2 --shards 2
+```
+
+### Where the code will live?
+
+The prometheus-operator-cli code will be placed in a dedicated repository under the [Prometheus organization](https://github.com/prometheus-operator), this will allow us to have a separeted lifecycle for the Prometheus Operator and the Prometheus Operator CLI and the following benefits:
```

Comment:
(nit)
```suggestion
The prometheus-operator-cli code will be placed in a dedicated repository under the [Prometheus organization](https://github.com/prometheus-operator), this will allow us to have a separate lifecycle for the Prometheus Operator and the Prometheus Operator CLI and the following benefits:
```